ivy/functional/ivy/general.py

```diff
@@ -1458,6 +1459,48 @@ def shape_to_tuple(shape: Union[ivy.Shape, ivy.NativeShape]):
     return tuple(shape)
 
 
+def to_ivy_shape(shape: Union[ivy.Shape, ivy.NativeShape]) -> ivy.Shape:
+    """Returns the input shape in ivy.Shape form
+
+    Parameters
+    ----------
+    shape
+        The input to be converted
+
+    Returns
+    -------
+     ret
+        the input in ivy.Shape form
+
+    """
+    return ivy.Shape(shape)
```

Comment:
we should only call this if `shape` is not an `ivy.Shape` already. I know the output will be the same regardless. But calls to `ivy.Shape` will be compiled into the graph (whereas `isinstance` checks will not), and therefore we should not add any more of these `ivy.Shape` calls into the computation stack than is necessary.